colcon_cmake/task/cmake/build.py

```diff
@@ -189,32 +190,49 @@ def _store_cmake_args(self, build_base, cmake_args):
     def _get_last_cmake_args_path(self, build_base):
         return Path(build_base) / 'cmake_args.last'
 
-    async def _build(self, args, env):
+    async def _build(self, args, env, *, additional_targets=None):
         self.progress('build')
 
         # invoke build step
         if CMAKE_EXECUTABLE is None:
             raise RuntimeError("Could not find 'cmake' executable")
-        cmd = [CMAKE_EXECUTABLE, '--build', args.build_base]
+
+        targets = []
         if args.cmake_target:
-            if args.cmake_target_skip_unavailable:
-                if not await has_target(args.build_base, args.cmake_target):
-                    return
-            cmd += ['--target', args.cmake_target]
-        if args.cmake_clean_first:
-            cmd += ['--clean-first']
-        if is_multi_configuration_generator(args.build_base, args.cmake_args):
-            cmd += ['--config', self._get_configuration(args)]
+            targets.append(args.cmake_target)
+        else:
+            targets.append('')
+            if additional_targets:
+                targets += additional_targets
 
+        multi_configuration_generator = is_multi_configuration_generator(
+            args.build_base, args.cmake_args)
+        if multi_configuration_generator:
             generator = get_generator(args.build_base)
             if 'Visual Studio' in generator:
                 env = self._get_msbuild_environment(args, env)
-        else:
-            job_args = self._get_make_arguments()
-            if job_args:
-                cmd += ['--'] + job_args
-        return await check_call(
-            self.context, cmd, cwd=args.build_base, env=env)
+
+        for i, target in enumerate(targets):
+            if args.cmake_target_skip_unavailable:
+                if not await has_target(args.build_base, target):
```

Comment:
How does this work if `target` is an empty string, which I'm assuming means "default targets", and `cmake_target_skip_unavailable` is `False`? I don't see anything in `has_target` for handling an empty string.

Reply:
Nice catch, that certainly does not work. Should be fixed in 17844894074d798144a54a25ebe9eb4f2fdc038f.